routes/owner.js

```diff
@@ -25,30 +26,42 @@ router.post('/hotel', async (req, res) => {
 
 router.put('/hotel/:id', async (req, res) => {
   const { error } = validate(req.body)
-  if (error) return res.status(400).send(error.details[0].message)
+  if (error) throw new ApiError(400, error.details[0].message)
   try {
     const hotel = await Hotel.findByIdAndUpdate(req.params.id, req.body)
 
     res.status(200).send(hotel)
   } catch (err) {
     console.log(err)
-    res.status(500).send('Something went wrong')
+    throw new ApiError(500, 'Something went wrong')
   }
 })
 
-router.delete('/hotel/:id', async (req, res) => {
+router.delete('/hotel/:id&:force', async (req, res) => {
   const id = req.params.id
+  const force = req.params.force
+  let message = ''
   try {
     const reservation = await Reservation.find({ hotelId: id })
 
-    if (reservation.length > 0)
-      return res.status(400).send('Remove reservations')
+    if (reservation.length > 0 && force === 'false')
+      throw new ApiError(
+        400,
+        'Remove reservations first or set flag force to true, please'
+      )
+
+    if (reservation.length > 0 && force === 'true') {
+      await Reservation.deleteMany({ hotelId: id })
+      message = 'and reservations '
+    }
 
     await Hotel.findByIdAndDelete(id)
-    res.status(200).send('Hotel deleted')
+
+    const hotels = await Hotel.find()
+    res.status(200).send({ message: `Hotel ${message}deleted`, hotels: hotels })
```

Comment:
Why do you return here list of hotels? I think we don't need it, in such situations when you change something I think is enough to return only information about success

Reply:
I had this in my issue

![image](https://user-images.githubusercontent.com/74857860/109210644-66a3f200-77ad-11eb-977c-c74478485162.png)